packages/calypso-products/src/camel-or-snake-product-type.ts

```diff
@@ -0,0 +1,7 @@
+import type { WithSnakeCaseProductType, WithCamelCaseProductType } from './types';
+
+export function camelOrSnakeProductType(
+	product: WithCamelCaseProductType | WithSnakeCaseProductType
+): string {
+	return 'product_type' in product ? product.product_type : product.productType;
```

Comment:
I'm not sure if it's actually necessary to support both, but the existing code for the product slug does the same thing, so I did it here too.

Reply:
There's something odd going on in calypso, where [underscore responses are mapped to snake case](https://github.com/Automattic/wp-calypso/blob/e7326458f8f0369c9f6c8a132c32b92b27265a08/client/lib/purchases/assembler.js#L77-L79), and [test data uses snake case](https://github.com/Automattic/wp-calypso/blob/c17f00d14ca19be8e25f485dca3b892b1a3d6058/client/state/themes/test/selectors.js#L2241), but my selectors checking those fields in that data [for snake case](https://github.com/Automattic/wp-calypso/blob/c17f00d14ca19be8e25f485dca3b892b1a3d6058/client/state/themes/selectors/get-purchased-themes.js#L17) work.
Could this be to support both untransformed and transformed objects?